### ISSUE TITLE
Plomino_Readers doesn't work when username different from userid

### ISSUE THREAD

**Issue by djay** [plomino/Plomino#830]
All the documentation seems to say that userid is what is to be stored in Plomino_Readers, but the code in both 1.x and 2.x searches based on username.

https://github.com/plomino/Plomino/blob/1.16.8/Products/CMFPlomino/index/PlominoIndex.py#L158

Is the fix search based on userid? @ebrehault

**Comment by ebrehault** [comment]
I remember @jean made some changes about that a long time ago, but we apparently missed that one.
We should use user id too.

**Comment by djay** [comment]
@ebrehault I'm just worried about backwards incompatibility. We could make it so it searches for either but its a gludge.

**Comment by ebrehault** [comment]
Yeah, I agre it can break compatibility, but still it seems the best option.

**Comment by djay** [comment]
PR coming today

**Comment by djay** [comment]
@ebrehault its a bit unclear what should happen with the special 'Anonymous' since it has a userid of None. Maybe it should at least have a userid that can't be used elsewhere in the system but there was already code that checked for `Anonymous` I think.
